butterfree/configs/db/s3_config.py

```diff
@@ -72,6 +72,30 @@ def get_options(self, key: str) -> dict:
             "path": os.path.join(f"s3a://{self.bucket}/", key),
         }
 
+    def get_path_with_partitions(self, key: str, dataframe: DataFrame) -> List:
+        """Get options for AWS S3 from partitioned parquet file.
+
+        Options will be a dictionary with the write and read configuration for
+        Spark to AWS S3.
+
+        Args:
+            key: path to save data into AWS S3 bucket.
+            dataframe: spark dataframe containing data from a feature set.
+
+        Returns:
+            A list of string for file-system backed data sources.
+        """
+        path_list = []
+        for row in dataframe.collect():
```

Comment:
What about using [this method](https://github.com/quintoandar/butterfree/blob/epic/incremental-feature/butterfree/dataframe_service/partitioning.py#L8) in order to extract partitions? What about the other options such as ```mode``` and ```format_```?

Reply:
Thank you, I forgot this method, I'm going to use it :smiley_cat: 
About the `mode` it is not necessary for the read method, since the `format_` is passed in another way, you can see it in the validate method in the HistoricalFeatureStoreWriter class.